pennylane/ops/qutrit/non_parametric_ops.py

```diff
@@ -0,0 +1,200 @@
+    The qutrit shift operator
+
+    .. math:: TShift = \begin{bmatrix}
+                        0 & 0 & 1 \\
+                        1 & 0 & 0 \\
+                        0 & 1 & 0
+                    \end{bmatrix}
+
+    **Details:**
+
+    * Number of wires: 1
+    * Number of parameters: 0
+
+    Args:
+        wires (Sequence[int] or int): the wire the operation acts on
+    """
+    num_wires = 1
+    """int: Number of wires that the operator acts on."""
+
+    num_params = 0
+    """int: Number of trainable parameters that the operator depends on."""
+
+    @staticmethod
+    def compute_matrix():
+        r"""Representation of the operator as a canonical matrix in the computational basis (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.TShift.matrix`
+
+        Returns:
+            ndarray: matrix
+
+        **Example**
+
+        >>> print(qml.TShift.compute_matrix())
+        [[0 0 1]
+         [1 0 0]
+         [0 1 0]]
+        """
+        return np.array([[0, 0, 1], [1, 0, 0], [0, 1, 0]])
+
+    @staticmethod
+    def compute_eigvals():
+        r"""Eigenvalues of the operator in the computational basis (static method).
+
+        If :attr:`diagonalizing_gates` are specified and implement a unitary :math:`U`,
+        the operator can be reconstructed as
+
+        .. math:: O = U \Sigma U^{\dagger},
+
+        where :math:`\Sigma` is the diagonal matrix containing the eigenvalues.
+
+        Otherwise, no particular order for the eigenvalues is guaranteed.
+
+        .. seealso:: :meth:`~.TShift.eigvals`
+
+        Returns:
+            array: eigenvalues
+
+        **Example**
+
+        >>> print(qml.TShift.compute_eigvals())
+        [ -0.5+0.8660254j -0.5-0.8660254j 1. +0.j         ]
+        """
+        return np.array([OMEGA, OMEGA**2, 1])
+
+    # TODO: Add compute_decomposition once parametric ops are added.
+
+    def pow(self, z):
+        if isinstance(z, int):
+            z_mod3 = z % 3
+            if z_mod3 < 2:
+                return super().pow(z_mod3)
+            return [self.adjoint()]
+        return super().pow(z)
+
+    def adjoint(self):
+        op = TShift(wires=self.wires)
+        op.inverse = not self.inverse
+        return op
+
+
+class TClock(Operation):
+    r"""TClock(wires)
+    Ternary Clock gate
+
+    .. math:: TClock = \begin{bmatrix}
+                        1 & 0      & 0        \\
+                        0 & \omega & 0        \\
+                        0 & 0      & \omega^2
+                    \end{bmatrix}
+                    \omega = \exp{2 \cdot \pi \cdot i / 3}
+
+    **Details:**
+
+    * Number of wires: 1
+    * Number of parameters: 0
+
+    Args:
+        wires (Sequence[int] or int): the wire the operation acts on
+    """
+    num_wires = 1
+    num_params = 0
```

Comment:
```suggestion
    num_wires = 1
    """int: Number of wires that the operator acts on."""

    num_params = 0
```
Just for consistency.